gwt-regexp/src/main/java/org/gwtproject/regexp/server/JavaRegExp.java

```diff
@@ -29,277 +30,275 @@
+            Pattern.compile("((?:^|\\G|[^$])(?:\\$\\$)*)\\$&");
+    private static final String REPLACEMENT_DOLLAR_AMPERSAND_FOR_JAVA = "$1\\$0";
+
+    // In JS syntax, a $` and $' in the replacement string stand for everything
+    // before the match and everything after the match.
+    // In Java syntax, there is no equivalent, so we detect and reject $` and $'.
+    // However, we have to watch out for $$` and $$', which are actually a JS $$
+    // (see below) followed by a ` or ' with no special meaning, and must not be
+    // rejected.
+    private static final Pattern REPLACEMENT_DOLLAR_APOSTROPHE =
+            Pattern.compile("(?:^|[^$])(?:\\$\\$)*\\$[`']");
+
+    // In JS syntax, a $$ in the replacement string stands for a (single) dollar
+    // sign, $.
+    // In Java syntax, the equivalent is \$, so we have to translate $$ to \$
+    // before passing it to Java.
+    private static final Pattern REPLACEMENT_DOLLAR_DOLLAR = Pattern.compile("\\$\\$");
+    // To get \$, we have to say \\\\\\$:
+    // \\\\\\$ --> Java string unescape --> \\\$
+    // \\\$ ---> Pattern replacement unescape in replacement preprocessing --> \$
+    private static final String REPLACEMENT_DOLLAR_DOLLAR_FOR_JAVA = "\\\\\\$";
+
+    /**
+     * Creates a regular expression object from a pattern with no flags.
+     *
+     * @param pattern the Javascript regular expression pattern to compile
+     * @return a new regular expression
+     * @throws RuntimeException if the pattern is invalid
+     */
+    public static RegExp compile(String pattern) {
+        return compile(pattern, "");
     }
 
-    Pattern javaPattern = Pattern.compile(pattern, javaPatternFlags);
-
-    return new JavaRegExp(pattern, javaPattern, globalFlag);
-  }
-
-  /**
-   * Returns a literal pattern <code>String</code> for the specified <code>String</code>.
-   *
-   * <p>
-   *
-   * <p>This method produces a <code>String</code> that can be used to create a <code>RegExp</code>
-   * that would match the string <code>s</code> as if it were a literal pattern. Metacharacters or
-   * escape sequences in the input sequence will be given no special meaning.
-   *
-   * @param input The string to be literalized
-   * @return A literal string replacement
-   */
-  public static String quote(String input) {
-    return Pattern.quote(input);
-  }
-
-  /**
-   * Parses a flags string as a set of characters. Does not reject unknown flags.
-   *
-   * @param flags the flag string to parse
-   * @return a set of flags
-   * @throws IllegalArgumentException if a flag is duplicated
-   */
-  private static Set<Character> parseFlags(String flags) {
-    Set<Character> flagsSet = new HashSet<Character>(flags.length());
-    for (int flagIndex = 0; flagIndex < flags.length(); flagIndex++) {
-      char flag = flags.charAt(flagIndex);
-      if (!flagsSet.add(flag)) {
-        throw new IllegalArgumentException("Flag cannot be specified twice: '" + flag + "'");
-      }
+    /**
```

Comment:
Is github just being dumb? It doesnt look like this file had big changes, but github is reporting that it did?

Reply:
Formatting only.